flax/linen/module.py

```diff
@@ -662,19 +662,32 @@ def __setattr__(self, name: str, val: Any):
       name: Attribute to set.
       val: Value of the attribute.
     """
-    is_dataclass_attr = name in self.__dataclass_fields__ and self.__dataclass_fields__[name].init  # pytype: disable=attribute-error
+    is_dataclass_attr = (
+      name in self.__dataclass_fields__ and  # pytype: disable=attribute-error
+      self.__dataclass_fields__[name].init)  # pytype: disable=attribute-error
```

Comment:
Maybe this is a bit nicer to read?

```suggestion
    fields = self.__dataclass_fields__
    is_dataclass_attr = (name in fields and fields[name].init)  # pytype: disable=attribute-error
```

Reply:
Definitely nicer!